src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -653,50 +661,51 @@ private static MethodDefinition createClientBeanMethod(
 
     bodyStatements.add(setBackgroundExecutorProviderStatement);
 
-    //   if (clientProperties.getUseRest()) {
-    //     clientSettingsBuilder.setTransportChannelProvider(
-    //         LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build());
-    //   }
-
-    MethodInvocationExpr getUseRest =
-        MethodInvocationExpr.builder()
-            .setMethodName("getUseRest")
-            .setReturnType(TypeNode.BOOLEAN)
-            .setExprReferenceExpr(thisClientPropertiesVarExpr)
-            .build();
-
-    // LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build()
-    Expr defaultTransportProviderBuider =
-        MethodInvocationExpr.builder()
-            .setStaticReferenceType(types.get("ServiceSettings"))
-            .setMethodName("defaultHttpJsonTransportProviderBuilder")
-            .build();
-    defaultTransportProviderBuider =
-        MethodInvocationExpr.builder()
-            .setExprReferenceExpr(defaultTransportProviderBuider)
-            .setMethodName("build")
-            .setReturnType(STATIC_TYPES.get("InstantiatingHttpJsonChannelProvider"))
-            .build();
-
-    MethodInvocationExpr setTransportProvider =
-        MethodInvocationExpr.builder()
-            .setExprReferenceExpr(VariableExpr.withVariable(settingBuilderVariable))
-            .setMethodName("setTransportChannelProvider")
-            .setArguments(defaultTransportProviderBuider)
-            .build();
-    IfStatement setTransportChannelProviderStatement =
-        createIfStatement(
-            getUseRest,
-            Arrays.asList(
-                ExprStatement.withExpr(setTransportProvider),
-                LoggerUtils.createLoggerStatement(
-                    ValueExpr.withValue(
-                        StringObjectValue.withValue("Using HTTP transport channel")),
-                    types)),
-            null);
-
-    bodyStatements.add(setTransportChannelProviderStatement);
-
+    if (hasRestOption) {
+      //   if (clientProperties.getUseRest()) {
+      //     clientSettingsBuilder.setTransportChannelProvider(
+      //         LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build());
+      //   }
+
+      MethodInvocationExpr getUseRest =
+          MethodInvocationExpr.builder()
+              .setMethodName("getUseRest")
+              .setReturnType(TypeNode.BOOLEAN)
+              .setExprReferenceExpr(thisClientPropertiesVarExpr)
+              .build();
+
+      // LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build()
+      Expr defaultTransportProviderBuider =
+          MethodInvocationExpr.builder()
+              .setStaticReferenceType(types.get("ServiceSettings"))
+              .setMethodName("defaultHttpJsonTransportProviderBuilder")
+              .build();
+      defaultTransportProviderBuider =
```

Comment:
Nit: consider naming this part something else instead of re-assigning to `defaultTransportProviderBuider` here? (Might help make parsing through the chained invocation logic a bit easier)